<?xml version="1.0"?>
<launch>
  <!-- Robot base -->
  <node pkg="jetbot_pro_ros2" exec="jetbot" name="jetbot" output="screen" respawn="true">
    <param name="port_name" value="/dev/ttyACM0"/>
    <param name="linear_correction" value="1.0"/>
    <param name="angular_correction" value="1.0"/>
    <remap from="/odom" to="/odom_raw"/>
    <param name="publish_odom_transform" value="false"/>
  </node>
  
  <node pkg="jetbot_pro_ros2" exec="odom_ekf" name="odom_ekf_node" output="screen"/>


  <!-- Start the GSCAM node -->
  <node pkg="gscam" exec="gscam_node" name="csi_cam_0" output="screen">
    <param name="camera_name" value="csi_cam_0"/>
    <param name="camera_info_url" value="package://jetbot_pro_ros2/config/camera_calibration/cam_640x480.yaml"/>
    <param name="gscam_config" value="nvarguscamerasrc sensor-id=0 ! video/x-raw(memory:NVMM),
    width=(int)640, height=(int)480, format=(string)NV12, framerate=(fraction)20/1 ! nvvidconv flip-method=0 ! videoconvert"/>  
    <param name="frame_id" value="/csi_cam_0_link"/>
    <param name="sync_sink" value="false"/>
    <remap from="camera/image_raw" to="csi_cam_0/image_raw"/>
    <remap from="/set_camera_info" to="csi_cam_0/set_camera_info"/>
  </node>


  <!-- LIDAR module -->
  <node name="rplidar_node" pkg="rplidar_ros" exec="rplidar_node" output="screen">
    <param name="serial_port" type="string" value="/dev/ttyACM1"/>
    <param name="serial_baudrate" type="int" value="115200"/><!--A1/A2 -->
    <!--param name="serial_baudrate" type="int" value="256000"--><!--A3 -->
    <param name="frame_id" type="string" value="laser_frame"/>
    <param name="inverted" type="bool" value="false"/>
    <param name="angle_compensate" type="bool" value="true"/>
  </node>


  <!-- Static transformations -->
	<node pkg="tf2_ros" exec="static_transform_publisher" name="base_footprint_to_imu" args="0 0 0.07 0 0 0 base_footprint base_imu_link 20"/>
  <node pkg="tf2_ros" exec="static_transform_publisher" name="base_link_to_laser" args="0 0.0 0.15 3.14 0.0 0.0 /base_footprint /laser_frame 20"/>
</launch>

<!--
	<node pkg="robot_pose_ekf" type="robot_pose_ekf" name="robot_pose_ekf" output="screen">
    <param name="output_frame" value="odom"/>
		<param name="base_footprint_frame" value="base_footprint"/>
    <param name="freq" value="30.0"/>
    <param name="sensor_timeout" value="0.5"/>
    <param name="odom_used" value="true"/>
    <param name="imu_used" value="true"/>
    <param name="vo_used" value="false"/>
    <param name="debug" value="false"/>
    <param name="self_diagnose" value="false"/>
    <remap from="odom" to="/odom_raw"/>
    <remap from="/imu_data" to="/imu"/>
		<remap from="/robot_pose_ekf/odom_combined" to="/odom_combined"/>
  </node>
-->
